libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -1285,12 +1283,10 @@ envFromPred env p = case p of
 -- | A version of `genFromSpecT` that simply errors if the generator fails
 genFromSpec :: forall fn a. (HasCallStack, HasSpec fn a) => Specification fn a -> Gen a
 genFromSpec spec = do
-  res <- strictGen $ explain1 "Calling genFromSpec" $ do
-    r <- genFromSpecT spec
-    unsafePerformIO $
-      r `seq`
-        pure (pure r) `catch` \(e :: SomeException) -> pure (fatalError (pure $ show e))
-  errorGE $ fmap pure res
+  res <- do
+    let genT = genFromSpecT @fn @a @GE spec
+    catchGen genT
```

Comment:
I am pretty sure that usage of `do` and `let` here makes the code less clear
```suggestion
  res <- catchGen $ genFromSpecT @fn @a @GE spec
```

Reply:
100%  ! Thanks